kv/kvserver: TestStoreCapacityAfterSplit failed

[(kv/kvserver).TestStoreCapacityAfterSplit failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2742525&tab=buildLog) on [master@99b543ec147a322e16a3f1e8850cd5575b6b049e](https://github.com/cockroachdb/cockroach/commits/99b543ec147a322e16a3f1e8850cd5575b6b049e):

```
=== RUN   TestStoreCapacityAfterSplit
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreCapacityAfterSplit280107970
    test_log_scope.go:74: use -show-logs to present logs inline
    client_split_test.go:2889: expected BytesPerReplica to have increased from p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00, but got p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00
    client_split_test.go:2934: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreCapacityAfterSplit280107970
--- FAIL: TestStoreCapacityAfterSplit (0.77s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestStoreCapacityAfterSplit PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStoreCapacityAfterSplit.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/client_split_test.go
+++ b/pkg/kv/kvserver/client_split_test.go
@@ -2833,7 +2833,7 @@ func TestStoreCapacityAfterSplit(t *testing.T) {
 	desc := tc.AddVotersOrFatal(t, key, tc.Target(1))
 	tc.TransferRangeLeaseOrFatal(t, desc, tc.Target(1))
 	testutils.SucceedsSoon(t, func() error {
-		repl, err := tc.GetFirstStoreFromServer(t, 1).GetReplica(desc.RangeID)
+		repl, err := s.GetReplica(desc.RangeID)
 		if err != nil {
 			return err
 		}
@@ -2843,7 +2843,7 @@ func TestStoreCapacityAfterSplit(t *testing.T) {
 		return nil
 	})
 
-	cap, err := s.Capacity(context.Background(), false /* useCached */)
+	cap, err := s.Capacity(ctx, false /* useCached */)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -2865,12 +2865,22 @@ func TestStoreCapacityAfterSplit(t *testing.T) {
 	// Increment the manual clock and do a write to increase the qps above zero.
 	manualClock.Increment(int64(kvserver.MinStatsDuration))
 	pArgs := incrementArgs(key, 10)
-	if _, pErr := kv.SendWrapped(context.Background(), s.TestSender(), pArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, s.TestSender(), pArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
-	tc.WaitForValues(t, key, []int64{10, 10})
+	// We want to make sure we can read the value through raft, so we know
+	// the stats are updated.
+	testutils.SucceedsSoon(t, func() error {
+		getArgs := getArgs(key)
+		if reply, err := kv.SendWrapped(ctx, s.TestSender(), getArgs); err != nil {
+			return errors.Errorf("failed to read data: %s", err)
+		} else if e, v := int64(10), mustGetInt(reply.(*roachpb.GetResponse).Value); v != e {
+			return errors.Errorf("failed to read correct data: expected %d, got %d", e, v)
+		}
+		return nil
+	})
 
-	cap, err = s.Capacity(context.Background(), false /* useCached */)
+	cap, err = s.Capacity(ctx, false /* useCached */)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -2904,11 +2914,11 @@ func TestStoreCapacityAfterSplit(t *testing.T) {
 
 	// Split the range to verify stats work properly with more than one range.
 	sArgs := adminSplitArgs(key.Next().Next())
-	if _, pErr := kv.SendWrapped(context.Background(), s.TestSender(), sArgs); pErr != nil {
+	if _, pErr := kv.SendWrapped(ctx, s.TestSender(), sArgs); pErr != nil {
 		t.Fatal(pErr)
 	}
 
-	cap, err = s.Capacity(context.Background(), false /* useCached */)
+	cap, err = s.Capacity(ctx, false /* useCached */)
 	if err != nil {
 		t.Fatal(err)
 	}

